### PR TITLE
Spr and spr2 64-bit interface

### PR DESCRIPTION
**Pull request by daineAMD**

### CODE REVIEW

File: clients/include/blas2/testing_spr.hpp [hunk 1]
```diff
@@ -70,50 +71,67 @@ void testing_spr_bad_arg(const Arguments& arg)
...
+    if(arg.bad_arg_all && arg.api & c_API_64)
+    {
+        hipblasLocalHandle handle(arg);
+        CHECK_HIPBLAS_ERROR(hipblasSetPointerMode(handle, HIPBLAS_POINTER_MODE_HOST));
+
+        int64_t n_64 = 2147483648; // will rollover to -2147483648 if using 32-bit interface
+        // rocBLAS implementation has alpha == 0 quick return after arg checks, so if we're using 32-bit params,
+        // this should fail with invalid-value
+        // Note that this strategy can't check incx as rocBLAS supports negative. Also depends on implementation so not testing cuBLAS for now
+        DAPI_CHECK(hipblasSprFn, (handle, uplo, n_64, &h_zero, nullptr, 1, nullptr));
```
Comment (TorreZuk) [inline clients/include/blas2/testing_spr.hpp]: Looks like good approach.  Add contants to yaml is better than magic.   You can still test on 32bit but use DAPI_EXPECT with conditional result, invalid_size : success to prove your test process itself and the constant.

**Review by TorreZuk — APPROVED** [review]
Suggestions made but otherwise looks good

File: clients/include/blas2/testing_spr2.hpp [hunk 2]
```diff
@@ -72,62 +73,73 @@ void testing_spr2_bad_arg(const Arguments& arg)
...
+    int64_t           incy = arg.incy;
 
-    int    abs_incx = incx < 0 ? -incx : incx;
-    int    abs_incy = incy < 0 ? -incy : incy;
-    size_t A_size   = size_t(N) * (N + 1) / 2;
-    size_t x_size   = abs_incx * size_t(N);
-    size_t y_size   = abs_incy * size_t(N);
+    int64_t abs_incx = incx < 0 ? -incx : incx;
+    int64_t abs_incy = incy < 0 ? -incy : incy;
+    size_t  A_size   = size_t(N) * (N + 1) / 2;
```
Comment (NaveenElumalaiAMD) [inline clients/include/blas2/testing_spr2.hpp:140]: size_t could be removed

File: clients/include/blas2/testing_spr2_batched.hpp [hunk 3]
```diff
@@ -75,99 +77,115 @@ void testing_spr2_batched_bad_arg(const Arguments& arg)
...
+                         batch_count));
+
+            int64_t n_64 = 2147483648; // will rollover to -2147483648 if using 32-bit interface
+            // rocBLAS implementation has alpha == 0 quick return after arg checks, so if we're using 32-bit params,
+            // this should fail with invalid-value
+            // Note that this strategy can't check incx as rocBLAS supports negative. Also depends on implementation so not testing cuBLAS for now
+            DAPI_EXPECT((arg.api & c_API_64) ? HIPBLAS_STATUS_SUCCESS
+                                             : HIPBLAS_STATUS_INVALID_VALUE,
+                        hipblasSpr2BatchedFn,
+                        (handle, uplo, n_64, zero, nullptr, 1, nullptr, 1, nullptr, n_64));
```
Comment (NaveenElumalaiAMD) [inline clients/include/blas2/testing_spr2_batched.hpp:176]: is batch_count intended to be n_64?

**Review by NaveenElumalaiAMD — APPROVED** [review]
LGTM  if all tests pass

File: clients/include/blas2/testing_spr2_batched.hpp (hunk 3 above)
Comment (daineAMD) [reply clients/include/blas2/testing_spr2_batched.hpp:176]: Yeah, might as well test batch_count has int64_t simultaneously

**Pull request merged**